ext/crypto/00_crypto.js

```diff
@@ -1185,6 +1278,52 @@
     }
   }
 
+  async function deriveBits(normalizedAlgorithm, baseKey, length) {
+    switch (normalizedAlgorithm.name) {
+      case "PBKDF2": {
+        // 1.
+        if (length == null || length == 0 || length % 8 !== 0) {
```

Comment:
Can you check if there's e.g. a WPT test that tests non-numeric inputs?

The logic is correct because e.g. `"bad" % 8` evaluates to `NaN` and `NaN !== 0` but I suspect that if it somehow managed to slip through to Rust land, it'd get coerced to zero.

Reply:
@bnoordhuis All the input types are always valid. They are validated by various webidl converters at the top of this function.

Reply:
Only [`null`](https://github.com/web-platform-tests/wpt/blob/03122ad70da015cc97a9bcd400f8eb83318d000d/WebCryptoAPI/derive_bits_keys/pbkdf2.js#L107) as non-numeric length is tested in WPT.

Wouldn't the WebIDL converter (`unsigned long`) also complain?

Reply:
Yes, it will. Strings will already fail parsing there, so will never reach this statement.